internal/core/services/rooms/room_manager.go

```diff
@@ -300,43 +305,53 @@ func (m *RoomManager) SchedulerMaxSurge(ctx context.Context, scheduler *entities
 }
 
 func (m *RoomManager) UpdateGameRoomStatus(ctx context.Context, schedulerId, gameRoomId string) error {
+	if _, err := m.updateGameRoomStatus(ctx, schedulerId, gameRoomId); err != nil {
```

Comment:
Isn't this the same as `return m.updateGameRoomStatus`? And in that case we could just dedup the functions and have just one `UpdateGameRooomStatus`